### PR TITLE
Niloofar/ Fix elements ts errors

### PR DESCRIPTION
**Pull request by niloo-fs**
Changes:

-   ...

## Type of change

-   [x] Bug fix
-   [ ] New feature
-   [ ] Update feature
-   [ ] Refactor code
-   [ ] Translation to code
-   [ ] Translation to crowdin
-   [ ] Script configuration
-   [ ] Improve performance
-   [ ] Style only
-   [ ] Dependency update
-   [ ] Documentation update
-   [ ] Release

### CODE REVIEW

File: src/components/elements/side-tab.tsx [hunk 1]
```diff
@@ -168,37 +168,38 @@ const SideTab = ({
...
-                            setMenu(!is_menu)
-                        }}
-                        active_tab={active_tab}
-                        label={label}
-                    />
-                </div>
-            )
-        })
-    }
+    const Tabs = () => (
```
Comment (farzin-deriv) [inline src/components/elements/side-tab.tsx:171]: @niloo-fs You should wrap `Tabs` with `memo`, Not `SideTab`.

File: src/components/elements/livechat-link-text.tsx [hunk 2]
```diff
@@ -2,22 +2,23 @@ import React from 'react'
...
+    const [is_livechat_interactive] = useLivechat()
 
     return (
         <LinkText
             color="red"
             mt="1rem"
             weight={weight}
             onClick={() => {
-                is_livechat_interactive && LC_API.open_chat_window()
+                is_livechat_interactive && window.LC_API.open_chat_window()
```
Comment (farzin-deriv) [inline src/components/elements/livechat-link-text.tsx:21]: @niloo-fs Why are you directly accessing `LC_API` from the `window` object? 🤔 
The idea behind getting it from the `useLivechat` hook was to hide the complexity and make it easer to change it in the future if needed, Not all the codebase need to know `LC_API` is coming from the `window`.

Comment (niloo-fs) [reply src/components/elements/livechat-link-text.tsx:21]: You told me that :))))

Comment (farzin-deriv) [reply src/components/elements/livechat-link-text.tsx:21]: @niloo-fs Let’s have a call on this whenever you are free 😅

File: src/components/elements/side-tab.tsx (hunk 1 above)
Comment (niloo-fs) [reply src/components/elements/side-tab.tsx:171]: My bad, will change this

File: src/components/elements/carousel/carousel-style.ts [hunk 3]
```diff
@@ -1,9 +1,27 @@
 import React from 'react'
 import styled, { css } from 'styled-components'
+import type { FlattenSimpleInterpolation } from 'styled-components'
 import { Flex } from 'components/containers'
 import device from 'themes/device'
 import { ReactComponent as Chevron } from 'images/svg/careers/carousel-chevron.svg'
 
+type StyledButtonWrapperProps = {
```
Comment (ali-hosseini-fs) [inline src/components/elements/carousel/carousel-style.ts:8]: based on the naming convention of deriv-com it should be `StyledButtonWrapperPropsType`, is that correct?

File: src/components/elements/carousel/carousel-style.ts [hunk 4]
```diff
@@ -1,9 +1,27 @@
...
+    left?: boolean
+}
+
+type NavigationContainerType = {
+    navigation_css?: FlattenSimpleInterpolation
+    bottom_offset?: string
+    height?: string
+}
+
+type ChevronRightAndLeftProps = {
```
Comment (ali-hosseini-fs) [inline src/components/elements/carousel/carousel-style.ts:20]: does this need `Type` in the end ?

File: src/components/elements/default-tab.tsx [hunk 5]
```diff
@@ -16,6 +16,11 @@ type TabsStyledProps = {
     selected?: boolean
 }
 
+type TextWrapperProps = {
```
Comment (ali-hosseini-fs) [inline src/components/elements/default-tab.tsx:19]: naming is correct?

File: src/components/elements/carousel/carousel-style.ts (hunk 3 above)
Comment (niloo-fs) [reply src/components/elements/carousel/carousel-style.ts:8]: No @ali-hosseini-fs as I checked they are using one of these 2 conventions: `ComponentNameProps` || `ComponentNameTypes`

File: src/components/elements/default-tab.tsx (hunk 5 above)
Comment (niloo-fs) [reply src/components/elements/default-tab.tsx:19]: yes @ali-hosseini-fs as I checked they are using one of these 2 conventions: `ComponentNameProps` || `ComponentNameTypes`

File: src/components/elements/carousel/carousel-style.ts (hunk 4 above)
Comment (niloo-fs) [reply src/components/elements/carousel/carousel-style.ts:20]: No @ali-hosseini-fs as I checked they are using one of these 2 conventions: ComponentNameProps || ComponentNameTypes

**Review by iman-fs — COMMENTED** [review]
there is another error in line 93 of `src/components/elements/auto-carousel.tsx`

image

can you check that please?

**Review by iman-fs — APPROVED** [review]

File: tsconfig.json [hunk 6]
```diff
@@ -4,7 +4,8 @@
     "compilerOptions": {
         "baseUrl": "src",
         "jsx": "preserve",
-        "lib": ["ES2021.String"],
+        "target": "ES6",
+        "lib": ["DOM", "DOM.Iterable", "ES6", "ES2021.String"],
```
Comment (ali-hosseini-fs) [inline tsconfig.json:8]: why do we need all of these?

Comment (niloo-fs) [reply tsconfig.json:8]: Thanks for checking, we don't need all of them, I'll remove extra options

**Review by ali-hosseini-fs — APPROVED** [review]

**Pull request merged**